msal/src/main/java/com/microsoft/identity/client/internal/controllers/BrokerMsalController.java

```diff
@@ -384,4 +444,66 @@ private Bundle getRequestBundleForRemoveAccount(@Nullable final IAccount account
 
         return requestBundle;
     }
+
+    /**
+     * Given an account, perform a global sign-out from this shared device (End my shift capability).
+     * This will invoke Broker and
+     * 1. Remove account from token cache.
+     * 2. Remove account from AccountManager.
+     * 3. Clear WebView cookies.
+     * 4. Sign out from default browser.
+     * */
+    public void removeAccountFromSharedDevice(@NonNull final IAccount account,
+                                              @NonNull final PublicClientApplicationConfiguration configuration,
+                                              @NonNull final IPublicClientApplication.AccountRemovedListener callback) {
+
+        sBackgroundExecutor.submit(new Runnable() {
+            @Override
+            public void run() {
+                IMicrosoftAuthService service;
+                final MicrosoftAuthClient client = new MicrosoftAuthClient(configuration.getAppContext());
+
+                try {
+                    final MicrosoftAuthServiceFuture authServiceFuture = client.connect();
+                    service = authServiceFuture.get();
+                    final Bundle requestBundle = getRequestBundleForGlobalSignOut(account);
+                    final Bundle resultBundle = service.removeAccountFromSharedDevice(requestBundle);
+
+                    Handler handler = new Handler(Looper.getMainLooper());
+                    handler.post(new Runnable() {
+                        @Override
+                        public void run() {
+                            if (resultBundle == null){
+                                callback.onAccountRemoved(true);
+                            } else {
+                                BrokerResult brokerResult = (BrokerResult) resultBundle.getSerializable(AuthenticationConstants.Broker.BROKER_RESULT_V2);
+                                com.microsoft.identity.common.internal.logging.Logger.error(
+                                    TAG,
+                                    "Failed to perform global sign-out."
+                                        + brokerResult.getErrorMessage(),
+                                    null);
+
+                                callback.onAccountRemoved(false);
+                            }
+                        }
+                    });
+                } catch (final BaseException | InterruptedException | ExecutionException | RemoteException e) {
+                    com.microsoft.identity.common.internal.logging.Logger.error(
+                        TAG,
+                        "Exception is thrown when trying to perform global sign-out."
+                            + e.getMessage(),
+                        e);
+                    callback.onAccountRemoved(false);
```

Comment:
Hmm... I wish we had some way to bubble the `Exception` up and tell why.... Maybe a future enhancement if the need arises

Reply:
For now I'm following the pattern we use for all binding functions, but I agree that bubbling this up might be helpful for devs.

Reply:
We can defer this for later. I imagine that if a substantial amount of customers hit this we'll be adding telemetry and more comprehensive error handling around it. I don't want this to be blocker - we can always capture device logs if need be